src/components/Signup/SignupForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { AuthError, signUp } from "aws-amplify/auth";
+import Icon from "components/Icon";
+import { Form } from "components/form";
+import { GENERIC_ERROR_MESSAGE } from "constants/common";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { requiredString } from "schemas/string";
+import { object } from "yup";
+import { Donor, StateSetter } from "./types";
+
+type Props = {
+  donor: Donor;
+  setSignupState: StateSetter;
+  classes?: string;
+};
+
+export default function SignupForm(props: Props) {
+  const { handleError } = useErrorContext();
+  const [isPasswordShown, setIsPasswordShown] = useState(false);
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+  } = useForm({
+    resolver: yupResolver(
+      object({
+        password: requiredString
+          .min(8, ({ min }) => `must have at least ${min} characters`)
+          .matches(/[a-z]/, "must have lowercase letters")
+          .matches(/[A-Z]/, "must have uppercase letters")
+          .matches(/\d/, "must have numbers")
+          .matches(
+            /[!@#$%^&*()_+{}\[\]:;<>,.?~\\/-]/,
```

Comment:
```suggestion
            /[! @#$%^&*()_+{}\[\]:;<>,.?~\\/-]/,
```
Should include space characters

Reply:
fixed